app/models/order.rb

```diff
@@ -0,0 +1,62 @@
+class Order < ApplicationRecord
+  include ActiveModel::Dirty
+
+  define_attribute_methods
+
+  has_many :line_items, dependent: :destroy, after_add: :calculate_price_and_discount_price_on_add, after_remove: :calculate_price_and_discount_price_on_remove, before_add: :prevent_multiple_purchase_of_same_deal, autosave: true
+  belongs_to :user
+  has_one :order_transaction, dependent: :destroy
+  has_many :deals, through: :line_items
+  belongs_to :shipping_address, class_name: 'Address', optional: true
+
+  enum status: [ :in_cart, :placed, :in_transit, :out_for_delivery, :shipped, :delivered, :cancelled ]
```

Comment:
let's use the gem state_machine to manage order's states
https://github.com/state-machines/state_machines